app/helpers/search_helper.rb

```diff
@@ -1,15 +1,19 @@
 # frozen_string_literal: true
 
 module SearchHelper
+  @@class_method_anchors = Concurrent::Map.new
+  @@instance_method_anchors = Concurrent::Map.new
```

Comment:
[Class variables are bad](https://rubystyle.guide/#no-class-vars). Can we use class instance variables? With getters, I guess.

Reply:
That comment is in reference to using them in a `class`, where this is a `module`. I personally don't know of a way to use a class variable in a `module`. `SearchHelper` gets included into a dynamically generated renderer class somewhere in Rails internals. I don't have a way to put class methods on that, to my knowledge. To your point, `AutocompleteResult` does also include `SearchHelper`, but is a shared cache not what we want? Why would we want to recompute the value in `AutocompleteResult`?

I personally did intend the cache to be shared across classes, which is one of the few times that a class variable actually is appropriate.